src/test/java/it/pagopa/ecommerce/commons/domain/v1/TransactionTest.java

```diff
@@ -3289,4 +3289,490 @@ void shouldConstructTransactionExpiredFromUserReceiptWithStatusNotifiedKO() {
+        );
+
+        TransactionActivated TransactionActivated = TransactionTestUtils
+                .transactionActivated(transactionActivatedEvent.getCreationDate());
+        TransactionWithRequestedAuthorization transactionWithRequestedAuthorization = TransactionTestUtils
+                .transactionWithRequestedAuthorization(authorizationRequestedEvent, TransactionActivated);
+        TransactionAuthorizationCompleted transactionAuthorizationCompleted = TransactionTestUtils
+                .transactionAuthorizationCompleted(
+                        authorizedEvent,
+                        transactionWithRequestedAuthorization
+                );
+        TransactionClosed transactionClosed = TransactionTestUtils
+                .transactionClosed(
+
+                        transactionAuthorizationCompleted,
+                        closureSentEvent
+                );
+
+        TransactionWithUserReceiptError transactionWithUserReceiptError = TransactionTestUtils
+                .transactionWithUserReceiptError(transactionClosed, userReceiptAddErrorEvent);
+
+        TransactionWithUserReceiptOk expected = TransactionTestUtils
+                .transactionWithUserReceiptOk(transactionWithUserReceiptError, userReceiptAddedEvent);
+        Mono<it.pagopa.ecommerce.commons.domain.v1.Transaction> actual = events
+                .reduce(transaction, it.pagopa.ecommerce.commons.domain.v1.Transaction::applyEvent);
+
+        StepVerifier.create(actual)
+                .expectNextMatches(
+                        t -> expected.equals(t)
+                                && (((TransactionWithUserReceiptOk) t).getStatus())
+                                        .equals(TransactionStatusDto.NOTIFIED_OK)
+                                && (((TransactionWithUserReceiptOk) t).getTransactionUserReceiptData()
+                                        .getResponseOutcome())
+                                                .equals(TransactionUserReceiptData.Outcome.OK)
+                )
+                .verifyComplete();
+    }
+
+    @Test
+    void shouldConstructTransactionWithAddedUserReceiptKoFromTransactionWithUserReceiptError() {
+        EmptyTransaction transaction = new EmptyTransaction();
+
+        TransactionActivatedEvent transactionActivatedEvent = TransactionTestUtils.transactionActivateEvent();
+        TransactionAuthorizationRequestedEvent authorizationRequestedEvent = TransactionTestUtils
+                .transactionAuthorizationRequestedEvent();
+        TransactionAuthorizationCompletedEvent authorizedEvent = TransactionTestUtils
+                .transactionAuthorizationCompletedEvent();
+        TransactionClosedEvent closureSentEvent = TransactionTestUtils
+                .transactionClosedEvent(TransactionClosureData.Outcome.OK);
+
+        TransactionUserReceiptAddErrorEvent userReceiptAddErrorEvent = TransactionTestUtils
+                .transactionUserReceiptAddErrorEvent(TransactionUserReceiptData.Outcome.OK);
```

Comment:
Shouldn't this be

```suggestion
                .transactionUserReceiptAddErrorEvent(TransactionUserReceiptData.Outcome.KO);
```
?